[GTK] Expose API to report theme color changes

aperezdc opened the pull request:
#### 4d7bd43827f50574456c8b329adabde9930411d6
<pre>
[GTK] Expose API to report theme color changes
<a href="https://bugs.webkit.org/show_bug.cgi?id=294389">https://bugs.webkit.org/show_bug.cgi?id=294389</a>

Reviewed by Carlos Garcia Campos.

Enable the WebKitWebView:theme-color property for the GTK port, based on
the implementation added for WPE in 296035@main. This mainly involved
removing PLATFORM(WPE) guards or adding PLATFORM(GTK) ones in code
common for both ports, and keeping some of the guards to use GdkRGBA
instead of WebKitColor to represent colors in the public API. The
existing test case is enabled for the GTK port, using the ColorType
alias type (which maps to GdkRGBA or WebKitColor depending on the port).
A small webkitWebViewEmitThemeColorChanged() private helper function was
also added, then changed the WPE port to make both ports use it.

Canonical link: <a href="https://commits.webkit.org/296160@main">https://commits.webkit.org/296160@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/41df1c86356a63577abf91bfa9072b451e3d6a0e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107510 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27192 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17601 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112726 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58046 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109474 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27880 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35693 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81615 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110439 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22085 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96896 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61999 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21523 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15036 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57490 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91453 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15068 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115826 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34577 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25501 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90651 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34952 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93151 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90397 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23047 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/rotated-cat-off-top-edge.html (failure)") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35313 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13088 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30334 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17395 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34498 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40063 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34245 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37600 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35905 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
```diff
@@ -1189,14 +1185,18 @@ static void webkitWebViewGetProperty(GObject* object, guint propId, GValue* valu
+        GdkRGBA color;
+        webkit_web_view_get_theme_color(webView, &color);
+        g_value_set_boxed(value, static_cast<gconstpointer>(&color));
+#endif
```
Very unimportant nit: it's semantically nicer to guard the GTK code with `PLATFORM(GTK)` guards, since theoretical future ports could use the WebKitColor code, but GdkRGBA will only ever be useful for GTK port.

aperezdc replied:
Sure thing, I'll change this to use `PLATFORM(GTK)` before landing 👌🏼

aperezdc committed:
[GTK] Expose API to report theme color changes
https://bugs.webkit.org/show_bug.cgi?id=294389

Reviewed by Carlos Garcia Campos.

Enable the WebKitWebView:theme-color property for the GTK port, based on
the implementation added for WPE in 296035@main. This mainly involved
removing PLATFORM(WPE) guards or adding PLATFORM(GTK) ones in code
common for both ports, and keeping some of the guards to use GdkRGBA
instead of WebKitColor to represent colors in the public API. The
existing test case is enabled for the GTK port, using the ColorType
alias type (which maps to GdkRGBA or WebKitColor depending on the port).
A small webkitWebViewEmitThemeColorChanged() private helper function was
also added, then changed the WPE port to make both ports use it.

Canonical link: https://commits.webkit.org/296160@main
